### PR TITLE
Correctly handle transfers with sender = receiver

### PR DESCRIPTION
**Pull request by arajasek**
And don't allow people to print money.

### CODE REVIEW

**Comment by arajasek** [comment]
Chain validation test demonstrating the fix https://github.com/filecoin-project/chain-validation/pull/210

File: chain/vm/vm.go [hunk 1]
```diff
@@ -607,16 +607,30 @@ func (vm *VM) transfer(from, to address.Address, amt types.BigInt) aerrors.Actor
...
+		return aerrors.Fatalf("transfer failed when resolving sender address: %s", err)
+	}
+
+	toId, err := vm.cstate.LookupID(to)
+	if err != nil {
+		return aerrors.Fatalf("transfer failed when resolving receiver address: %s", err)
+	}
+
+	if fromId == toId {
+		return nil
```
Comment (magik6k) [inline chain/vm/vm.go:621]: Will this fail when there are not enough funds to transfer? I guess yes because there are some message checks earlier right?

Comment (arajasek) [reply chain/vm/vm.go:621]: Yeah, that should be caught at the beginning of `ApplyMessage`...Might be worth adding a redundant check anyway for internal messages.

**Comment by arajasek** [comment]
> Maybe more of a spec than an implementation question: is there a case where a message with the same actor as both receiver and sender can be valid? Same question for an _external_ message.

I'd say, yeah, they're always valid, if that's what the actor chose to do for some reason? Assuming they meet all the other validity checks.

**Comment by arajasek** [comment]
Needs to be rebased after #1856 goes in, since chain-val already depends on that

**Pull request merged**